File: 0997-find-the-town-judge-找到小镇的法官/0997.py
class Solution:
    def findJudge(self, N: int, trust: List[List[int]]) -> int:
        if len(trust) == 0: 
            if N == 1: return 1
            else: return -1
        people = {}
        for t in trust:
            if t[1] in people:
                people[t[1]]["left"].append(t[0])
            else:
                people[t[1]] = {"left":[t[0]], "right":[]}
            if t[0] in people:
                people[t[0]]["right"].append(t[1])
            else:
                people[t[0]] = {"left":[], "right":[t[1]]}
        
        for index, person in people.items():
            if len(person["right"]) == 0 and len(person["left"]) == N - 1:
                return index
        
        return -1
            

File: 0997-find-the-town-judge-找到小镇的法官/0997.cpp
class Solution {
public:
    int findJudge(int N, vector<vector<int>>& trust) {
        int act[N] = {0}; // 该人act[i] 相信的人的数量
        int pas[N] = {0}; // 该人act[i] 被多少人相信

        for(auto &p: trust) {
            act[p[0] - 1] += 1;
            pas[p[1] - 1] += 1;
        }

        for(int i = 0; i != N; ++i) {
            if (act[i] == 0 && pas[i] == N - 1) return i + 1;
        }
        return -1;
    }
};